pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -736,3 +865,11 @@ func (*mvccGCQueue) purgatoryChan() <-chan time.Time {
 func (*mvccGCQueue) updateChan() <-chan time.Time {
 	return nil
 }
+
+func checkRelicaIsLeaseHolder(ctx context.Context, replica *Replica, timeout time.Duration) error {
+	return contextutil.RunWithTimeout(ctx, "check-replica-lease", timeout,
+		func(ctx context.Context) error {
+			_, err := replica.redirectOnOrAcquireLease(ctx)
```

Comment:
Is there a reason why we use `redirectOnOrAcquireLease` here rather than the check that the `baseQueue` uses (which doesn't try to acquire the lease)?

https://github.com/cockroachdb/cockroach/blob/fc12919071a37f659d9bfdb0c8ba16bb9b46cd41/pkg/kv/kvserver/queue.go#L677-L687

Reply:
My concern was that if deleted range gets invalid lease it would not be picked up. But I see that we only acquire leaseholder when actually processing, so this is probably fine to change here to just check valid leases.